backend/typescript/rest/schedulingRoutes.ts

```diff
@@ -203,18 +203,51 @@ schedulingRouter.post("/", createSchedulingDtoValidator, async (req, res) => {
   }
 });
 
-/* Update the scheduling instance by id */
-schedulingRouter.put("/:id", updateSchedulingDtoValidator, async (req, res) => {
-  try {
-    const updatedScheduling = await schedulingService.updateSchedulingById(
-      req.params.id,
-      req.body,
-    );
-    res.status(200).json(updatedScheduling);
-  } catch (error: unknown) {
-    res.status(500).json({ error: getErrorMessage(error) });
-  }
-});
+/* Update the scheduling instance by id or updates by recurring donation id  */
+schedulingRouter.put(
+  "/:id?",
+  updateSchedulingDtoValidator,
+  async (req, res) => {
+    const { id } = req.params;
+    const { recurringDonationId } = req.query;
+    const contentType = req.headers["content-type"];
+
+    if (recurringDonationId && id) {
+      await sendResponseByMimeType(res, 400, contentType, [
+        {
+          error: "Cannot delete by both id and recurringDonationId",
+        },
+      ]);
+      return;
+    }
+    if (id) {
+      try {
+        const updatedScheduling = await schedulingService.updateSchedulingById(
+          id,
+          req.body,
+        );
+        res.status(200).json(updatedScheduling);
+      } catch (error: unknown) {
+        res.status(500).json({ error: getErrorMessage(error) });
+      }
+    }
+    if (recurringDonationId) {
```

Comment:
this should be else if, or else it'll throw the "Must supply id or recurringDonationId" error when it's an id as well

Reply:
that wont happen cuz if (id) is above that if statement, but I should have else if anyways